### DISCUSSION TITLE
Auto Discover Repos with specific file in root directory

### DISCUSSION THREAD

**Discussion by capeflynn** [renovatebot/renovate#25067] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

latest

### If you're self-hosting Renovate, select which platform you are using.

Bitbucket Cloud (bitbucket.org)

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I would like to only scan repos that already have a `renovate.json` file in it or are part of a Bitbucket Cloud project. Is there anyway to filter Bitbucket Cloud repos by project?

### What you tried so far.

I've tried using the `autodiscoverFilter` for this purpose, but I cannot get it to work. It doesn't seem to recognize anything below the repository level.

This works...
```
autodiscoverFilter: ["<workspace>/<repo-name>"]
```

but this does not...
```
autodiscoverFilter: ["<workspace>/**/renovate.json"]
```

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment, marked as answer]
To scan only repos with an existing renovate.json, you can set onboarding=false. Repos without renovate.json will be skipped.

You can't achieve that _or_ matching a project - you'd need to run Renovate twice with two different settings.

**Reply by capeflynn** [reply]
Hi @rarkins, thank you again for your help. This answered my question.

**Comment by capeflynn** [comment]
Hi @rarkins,

Thank you so much for the quick response! I saw this option, but I'm not sure it will work for us. One of the caveats listed when using this setting is that the `renovate.json` file must be empty, but we had to implement the solution mentioned in the [Update versions defined in a property but managed by parent pom](https://github.com/renovatebot/renovate/issues/15170#issuecomment-1287307080) issue. I could not figure out how to add the regex manager to the global `config.js` file, so it had to live in each repos' `renovate.json` file.

```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "regexManagers": [{
    "fileMatch": ["(^|/|\\.)pom\\.xml$"],
    "matchStrings": ["<\\S+\\.version renovate:depName=\"(?<depName>\\S+?)\">(?<currentValue>.+?)<\\/\\S+\\.version>"],
    "versioningTemplate": "maven",
    "datasourceTemplate": "maven"
  }]
}
```

**Reply by rarkins** [reply]
The file doesn't need to be empty

**Reply by capeflynn** [reply]
OK, great. I will give it a try, thank you.

Just in case it is helpful, this is where I thought it said the file needed to be empty...https://docs.renovatebot.com/self-hosted-configuration/#onboarding

![image](https://github.com/renovatebot/renovate/assets/50003041/2d537254-f196-4133-9806-00e821bcc5bd)
